libs/clients/syslumenn/src/lib/syslumennClient.service.ts

```diff
@@ -382,83 +399,135 @@ export class SyslumennService {
-    propertyNumber: string,
-  ): Promise<MortgageCertificate> {
+    properties: {
+      propertyNumber: string
+      propertyType: string
+    }[],
+  ): Promise<MortgageCertificate[]> {
     const { id, api } = await this.createApi()
 
-    const res = await api.vedbokarvottordPost({
+    const res = await api.vedbokarvottord2Post({
       skilabod: {
         audkenni: id,
-        fastanumer: cleanPropertyNumber(propertyNumber),
-        tegundAndlags: VedbondTegundAndlags.NUMBER_0, // 0 = Real estate
+        eignir: properties.map(({ propertyNumber, propertyType }) => {
+          return {
+            fastanumer: propertyNumber,
+            tegundAndlags:
+              propertyType === '1'
+                ? VedbondTegundAndlags.NUMBER_1 // 1 = Vehicle
+                : propertyType === '2'
+                ? VedbondTegundAndlags.NUMBER_2 // 2 = Ship
+                : VedbondTegundAndlags.NUMBER_0, // 0 = Real estate
+          }
+        }),
       },
     })
-    const contentBase64 = res.vedbandayfirlitPDFSkra || ''
 
-    const certificate: MortgageCertificate = {
-      contentBase64: contentBase64,
-      apiMessage: res.skilabod,
-    }
+    const certificates: MortgageCertificate[] =
+      res.skilabodOgSkra?.map((resultItem) => {
+        const contentBase64 = resultItem.vedbandayfirlitPDFSkra || ''
+        return {
+          contentBase64: contentBase64,
+          apiMessage: resultItem.skilabod,
+          propertyNumber: resultItem.fastanumer,
+        }
+      }) ?? []
 
-    return certificate
+    return certificates
   }
 
   async validateMortgageCertificate(
-    propertyNumber: string,
-    isFromSearch: boolean | undefined,
-  ): Promise<MortgageCertificateValidation> {
+    properties: {
+      propertyNumber: string
+      propertyType: string
+    }[],
+  ): Promise<MortgageCertificateValidation[]> {
     try {
       // Note: this function will throw an error if something goes wrong
-      const certificate = await this.getMortgageCertificate(propertyNumber)
-
-      const exists = certificate.contentBase64.length !== 0
-      const hasKMarking =
-        exists && certificate.contentBase64 !== 'Precondition Required'
+      const certificates = await this.getMortgageCertificate(properties)
 
       // Note: we are saving propertyNumber and isFromSearch also in externalData,
       // since it is not saved in answers if we go from state DRAFT -> DRAFT
-      return {
-        propertyNumber: propertyNumber,
-        isFromSearch: isFromSearch,
-        exists: exists,
-        hasKMarking: hasKMarking,
-      }
+      return certificates.map((certificate) => {
+        const exists = certificate.contentBase64.length !== 0
+        const hasKMarking =
+          exists && certificate.contentBase64 !== 'Precondition Required'
+
+        return {
+          propertyNumber: certificate.propertyNumber ?? '',
+          exists: exists,
+          hasKMarking: hasKMarking,
+        }
+      })
```

Comment:
Looks like the syslumennClient.utils hosts mappers. Maybe consider moving some mappers there to maintain the structure?

Reply:
Ah, I see you're already making some mappers so maybe this one isn't applicable.

Reply:
No, you are right, it would definitely look cleaner in a mapper